pandas/core/dtypes/cast.py

```diff
@@ -159,6 +159,31 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+# TODO: this should be a TypeVar
```

Comment:
can you remove the use of maybe_box_datetimelike with this instead, or rather just incorporate it here

Reply:
I can but I think we can't eliminate `maybe_box_datetimelike` atm because we use it in
https://github.com/pandas-dev/pandas/blob/07559156ba09adfa1193bc9a88e3bf38c9146857/pandas/core/indexes/interval.py#L1197-L1198
and in
https://github.com/pandas-dev/pandas/blob/07559156ba09adfa1193bc9a88e3bf38c9146857/pandas/core/internals/blocks.py#L848
where AFAICT it shouldn't be replaced with `maybe_box_native` (or if it can I'd like to do that separately, in a follow-on)